retriever/mongodbretriever/retriever.go

```diff
@@ -0,0 +1,78 @@
+package mongodbretriever
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+
+// Retriever is a configuration struct for a MongoDB connection and Collection.
+type Retriever struct {
+	Uri          string
+	Collection   string
+	Database     string
+	dbConnection *mongo.Database
+}
+
+// Retrieve is reading flag configuration from mongodb collection and returning it
+func (r *Retriever) Retrieve(ctx context.Context) ([]byte, error) {
+
+	if r.dbConnection == nil {
+
+		client, err := mongo.Connect(ctx, options.Client().ApplyURI(r.Uri))
+		if err != nil {
+			return nil, err
+		}
+		r.dbConnection = client.Database(r.Database)
+	}
+	//TODO: how to gracefully disconnect?
+	// defer func () {
+	// 	if err := client.Disconnect(ctx); err != nil {
+	// 		panic(err)
+	// 	}
+	// }()
```

Comment:
I had a bit of time tonight and have made this PR #1427, I think it can solve your issue with `Init` and `Shutdown` and make it easier for you to support the connections.
Do you mind having a look and tell me if it helps?

If you want to understand how it works, you can check the preview of the doc [here](https://6595e4791c74be00079fb4f8--go-feature-flag-doc-preview.netlify.app/docs/next/go_module/store_file/custom#initializable-retriever).